hikari/auto_mod.py

```diff
@@ -0,0 +1,231 @@
+# cython: language_level=3
+# Copyright (c) 2020 Nekokatt
+# Copyright (c) 2021-present davfsa
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+"""Entities that are used to describe auto-moderation on Discord."""
+
+from __future__ import annotations
+
+__all__: typing.Sequence[str] = (
+    "AutoModActionType",
+    "PartialAutoModAction",
+    "AutoModBlockMessage",
+    "AutoModSendAlertMessage",
+    "AutoModTimeout",
+    "AutoModEventType",
+    "AutoModTriggerType",
+    "AutoModKeywordPresetType",
+    "PartialAutoModTrigger",
+    "KeywordTrigger",
+    "HarmfulLinkTrigger",
+    "SpamTrigger",
+    "KeywordPresetTrigger",
+    "AutoModRule",
+)
+
+import typing
+
+import attrs
+
+from hikari import snowflakes
+from hikari.internal import attrs_extensions
+from hikari.internal import enums
+
+if typing.TYPE_CHECKING:
+    import datetime
+
+    from hikari import traits
+
+
+class AutoModActionType(int, enums.Enum):
+    """The type of an auto-moderation rule action."""
+
+    BLOCK_MESSAGES = 1
+    """Block the content of the triggering message."""
+
+    SEND_ALERT_MESSAGE = 2
+    """Log the triggering user content to a specified channel."""
+
+    TIMEOUT = 3
+    """Timeout the triggering message's author for a specified duration.
+
+    This type can only be set for `KEYWORD` rules and requires the `MODERATE_MEMBERS`
+    permission to use.
+    """
+
+
+@attrs_extensions.with_copy
+@attrs.define(kw_only=True, weakref_slot=False)
+class PartialAutoModAction:
```

Comment:
https://github.com/hikari-py/hikari/pull/1220#discussion_r1122617484 if this is the case this would be an easy change